pandas/io/formats/format.py

```diff
@@ -1273,6 +1273,11 @@ def format_percentiles(percentiles):
 
 def _is_dates_only(values):
     # return a boolean if we are only dates (and don't have a timezone)
+    if isinstance(values, np.ndarray) and values.ndim > 1:
```

Comment:
In which case do you run into this? 
(I was assuming the format_array is working column by column)

Reply:
Per above, this is hit with 2D ndarray inputs, which ATM are incorrectly accepted but will now raise.

Reply:
Yes, but my question is: *when* is this actually hit with 2D ndarray input?

Reply:
same question as @jorisvandenbossche

Reply:
Just added an assertion for 1D-ness (in master) and the first test that failed is effectively:

`pd.DataFrame({"A": pd.date_range('2016-01-01', periods=3)}).to_csv()`

Reply:
OK. I would rather do a ravel in the code calling it:

https://github.com/pandas-dev/pandas/blob/f0919f272d9614058b5ebb5e0664d1ac6f23540f/pandas/core/internals/blocks.py#L2171-L2175

as it is also done for the actual formatting function right below. 

In fact, this is also kind of a bug in our formatting. As the formatting should be done column by column (the frequency of one column should not influence the formatting of another column)

Reply:
OK.  I think is_dates_only is only called with non-ravelled data in one place, so I can move the maybe ravel there and put an assertion in is_dates_only.  Is there anything else that should go along with that?

Reply:
did you update this?